### PR TITLE
[rayci] support arm64 wanda and test

### PR DESCRIPTION
**Pull request by aslonnie**
also adds platform in wanda core to distinguish containers built for different platforms

### CODE REVIEW

**Comment by aslonnie** [comment]
https://buildkite.com/ray-project/rayci/builds/477#018bb228-e2e7-4eb2-b696-d096b9d087a3

works pretty good

**Comment by aslonnie** [comment]
just fixed a x-platform cross talking bug.

File: run_rayci.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ TMP_DIR="$(mktemp -d)"
 
 echo "--- Install rayci"
 
-curl -sL 'https://go.dev/dl/go1.20.6.linux-amd64.tar.gz' | tar -xzf - -C "$TMP_DIR"
+curl -sfL 'https://go.dev/dl/go1.21.4.linux-amd64.tar.gz' | tar -xzf - -C "$TMP_DIR"
```
Comment (can-anyscale) [inline run_rayci.sh:11]: where is this script used? does it need to support arm as well?

**Review by can-anyscale — APPROVED** [review]

File: run_rayci.sh (hunk 1 above)
Comment (aslonnie) [reply run_rayci.sh:11]: rayci and this script runs only on the very first step, for bootstrapping, so this does not need to support arm64 (or at least not important).

wanda needs to support arm64.

**Pull request merged**